Lib/test/test_zipfile/test_core.py

```diff
@@ -3460,40 +3460,30 @@ def read(self, size=-1):
                 self.bytes_read += len(bs)
                 return bs
 
-            def get_bytes_read(self):
-                return self.bytes_read
-
         sio = StatIO()
-        # 100000 bytes
-        txt = b'0123456789'*10000
+        # 20000 bytes
+        txt = b'0123456789' * 2000
 
-        # Check seek on a file
         with zipfile.ZipFile(sio, "w", compression=zipfile.ZIP_STORED) as zipf:
             zipf.writestr("foo.txt", txt)
 
+        # check random seek and read on a file
         with zipfile.ZipFile(sio, "r") as zipf:
             with zipf.open("foo.txt", "r") as fp:
-                br = sio.get_bytes_read()
-                fp.seek(50000, os.SEEK_CUR)
-                self.assertEqual(sio.get_bytes_read() - br, 0, 'seek produces redundant read!')
-
-                b = fp.read(100)
-                self.assertEqual(b, txt[:100])
-
                 # seek length must be greater than ZipExtFile.MIN_READ_SIZE (4096)
-                # backward seek
-                br = sio.get_bytes_read()
-                fp.seek(5000, os.SEEK_CUR)
-                b = fp.read(100)
-                self.assertEqual(b, txt[50000:50100])
-                self.assertLessEqual(sio.get_bytes_read() - br, 4096, 'read redundant bytes during backward seek!')
-
                 # forward seek
-                br = sio.get_bytes_read()
-                fp.seek(-40000, os.SEEK_CUR)
-                b = fp.read(100)
-                self.assertEqual(b, txt[10100:10200])
-                self.assertLessEqual(sio.get_bytes_read() - br, 4096, 'read redundant bytes during forward seek!')
+                old_count = sio.bytes_read
+                fp.seek(10002, os.SEEK_CUR)
+                arr = fp.read(100)
+                self.assertEqual(arr, txt[10002:10102])
+                self.assertLessEqual(sio.bytes_read - old_count, 4096, 'Redundant bytes were read during forward seek and read!')
+
+                # backward seek
+                old_count = sio.bytes_read
+                fp.seek(-5003, os.SEEK_CUR)
```

Comment:
Just in case, whenever you seek or read, you can add an assertion `fp.tell() == ...` to check the current position. You should hardcode the position to check though.